src/Client/ConcordiumClient.cs

```diff
@@ -545,6 +549,112 @@ await Task.WhenAll(response.ResponseHeadersAsync, response.ResponseAsync)
             .ConfigureAwait(false);
     }
 
+    /// <summary>
+    /// Return a stream of blocks that arrive from the time the query is made onward.
+    /// This can be used to listen for incoming blocks.
+    /// </summary>
+    /// <param name="token">Cancellation token</param>
+    /// <returns>A stream of blocks that arrive from the time the query is made onward.</returns>
+    /// <exception cref="RpcException">
+    /// RPC error occurred, access <see cref="RpcException.StatusCode"/> for more information.
+    /// <see cref="StatusCode.Unimplemented"/> indicates that this endpoint is disabled in the node.
+    /// </exception>
+    public IAsyncEnumerable<ArrivedBlockInfo> GetBlocks(CancellationToken token = default)
+    {
+        var response = this.Raw.GetBlocks(token);
+        return response.ResponseStream.ReadAllAsync(token).Select(ArrivedBlockInfo.From);
+    }
+
+    /// <summary>
+    /// Return a stream of blocks that are finalized from the time the query is
+    /// made onward. This can be used to listen for newly finalized blocks. Note
+    /// that there is no guarantee that blocks will not be skipped if the client is
+    /// too slow in processing the stream, however blocks will always be sent by
+    /// increasing block height.
+    /// </summary>
+    /// <param name="token">Cancellation token</param>
+    /// <returns>A stream of finalized blocks that arrive from the time the query is made onward.</returns>
+    /// <exception cref="RpcException">
+    /// RPC error occurred, access <see cref="RpcException.StatusCode"/> for more information.
+    /// <see cref="StatusCode.Unimplemented"/> indicates that this endpoint is disabled in the node.
+    /// </exception>
+    public IAsyncEnumerable<FinalizedBlockInfo> GetFinalizedBlocks(CancellationToken token = default)
+    {
+        var response = this.Raw.GetFinalizedBlocks(token);
+        return response.ResponseStream.ReadAllAsync(token).Select(FinalizedBlockInfo.From);
+    }
+
+    /// <summary>
+    /// Get the current branches of blocks starting from and including the last finalized block.
+    /// </summary>
+    /// <param name="token">Cancellation token</param>
+    /// <returns>The current branches of blocks.</returns>
+    /// <exception cref="RpcException">
+    /// RPC error occurred, access <see cref="RpcException.StatusCode"/> for more information.
+    /// <see cref="StatusCode.Unimplemented"/> indicates that this endpoint is disabled in the node.
+    /// </exception>
+    public async Task<Branch> GetBranchesAsync(CancellationToken token = default)
+    {
+        var response = await this.Raw.GetBranchesAsync(token);
```

Comment:
It is good practice for libraries to add `.ConfigureAwait(false)` when awaiting tasks:
```suggestion
        var response = await this.Raw.GetBranchesAsync(token).ConfigureAwait(false);
```